core/src/main/kotlin/knote/script/NotebookConfiguration.kt

```diff
@@ -35,76 +39,35 @@ class NotebookConfiguration : ScriptCompilationConfiguration({
 
         beforeCompiling { context ->
             val reports = mutableListOf<ScriptDiagnostic>()
+
+            val scriptFile = (context.script as FileScriptSource).file
+            val id = scriptFile.name.substringBeforeLast(".notebook.kts")
+
             reports += ScriptDiagnostic(
                 "beforeCompiling time: ${System.currentTimeMillis()}",
                 ScriptDiagnostic.Severity.DEBUG
             )
 
+            reports += ScriptDiagnostic(
+                "knote version: ${CoreConstants.FULL_VERSION}",
+                ScriptDiagnostic.Severity.INFO
```

Comment:
Now I'm seeing so many of these, these are SO important, but they do add a lot to the code that sort of takes away focus from what it actually does.


I think I am interested in creating DSLs for making shorter code for these Script Diagnostics. That's what I'm interested in creating another PR for (another personal note to myself). Maybe I'll do a prototype tonight for this

Reply:
as in a extension function for making diagnostics? that would be nice
something like `reports += diagnostic.info("knote version: ${CoreConstants.FULL_VERSION}")`
would be a lot better already

Reply:
yeah, definitely. I'll start a new branch when we merge this in so we can try to make all that less in the way of your important code!